apps/viewer/uicallbacks.js

```diff
@@ -2561,62 +2570,52 @@ async function rootCallback({root, parent, items}) {
-  document.querySelector('[title="Enhance"]').previousSibling.checked = true;
-  if (!setEnhance) {
-    // $UI.message.add('<i class="small material-icons">info</i>On Movement, enhance would be undone', 2000);
-    $CAMIC.viewer.addHandler('zoom', unenhance);
-    $CAMIC.viewer.addHandler('pan', unenhance);
-    setEnhance = true;
-  }
-
-  // Canvas information
-  const canvas = $CAMIC.viewer.canvas.firstChild;
-  const context = canvas.getContext('2d');
-  let width = canvas.width; let height = canvas.height;
-  var img = context.getImageData(0, 0, width, height);
-
-  if (data.status == 'Histogram Eq') {
-    context.putImageData(clahe(img, 64, 0.015), 0, 0);
-  } else if (data.status == 'Edge') {
-    context.putImageData(edgedetect(img, 150, 0, 0, width, height), 0, 0);
-  } else if (data.status == 'Sharpen') {
-    var filter = [[-1, -1, -1], [-1, 14, -1], [-1, -1, -1]];
-    filter = [filter, filter, filter];
-    var newimg = new ImageData(new Uint8ClampedArray(applyfilter(img, filter, 0, 0, width, height)), width, height);
-    context.putImageData(newimg, 0, 0);
-  } else if (data.status == 'Custom') {
-    var input = prompt('Enter the 2D Kernel: Eg : [[1, 0], [0, 1]]'); var f=0;
-    // JSON Parse test
-    try {
-      var filter = JSON.parse(input); var sz = filter[0].length;
-    } catch (e) {
-      console.error(e);
-      alert('Invalid Kernel : ' + input);
-      return;
-    }
-    // 2D array check
-    for (var r=0; r<filter.length; r++) {
-      if (!Array.isArray(filter[r]) || filter[r].length!=sz || sz==0) f=1;
-      else {
-        for (var c = 0; c<filter[r].length; c++) {
-          if (Array.isArray(filter[r][c]))f=1;
-        }
+function enhance(e){
+    document.querySelector('[title="Enhance"]').previousSibling.checked = true;
+    if(!set_enhance){
+      // $UI.message.add('<i class="small material-icons">info</i>On Movement, enhance would be undone', 2000);
+      $CAMIC.viewer.addHandler('zoom', unenhance);
+      $CAMIC.viewer.addHandler('pan', unenhance);
+      set_enhance = true;}
+
+    // Canvas information
+    const canvas = $CAMIC.viewer.canvas.firstChild;
+    const context = canvas.getContext('2d');
+    let width = canvas.width, height = canvas.height;
+    var img = context.getImageData(0,0,width,height);
+    var data = img.data;
+
+    if (e.status == 'Histogram Eq')
+        context.putImageData(clahe(img,64,0.015),0,0);
+    else if (e.status == 'Edge'){
+        context.putImageData(edgedetect_sobel(img,80),0,0);}
+    else if(e.status == 'Sharpen'){
+        var filter = [[-1,-1,-1],[-1,14,-1],[-1,-1,-1]];
+        context.putImageData(applyfilter(img,filter),0,0);}
+
+    else if (e.status == "Custom"){
+        var input = prompt("Enter the 2D Kernel: Eg : [[1, 0], [0, 1]]"), f=0;
+        // JSON Parse test
+        try{
+        var filter = JSON.parse(input), sz = filter[0].length;}
+        catch{
+          alert("Invalid Kernel : " + input);
+          return;}
```

Comment:
I have a lil issue with the placing of the closing curly brace `}`, everything else looks good!